packages/side-nav/src/SideNavItem.tsx

```diff
@@ -180,34 +163,39 @@ export interface SideNavItemProps {
  @param props.as When provided, the component will be rendered as the component or html tag indicated
  *  by this prop. Other additional props will be spread on the anchor element.
  */
-
 const SideNavItem: ExtendableBox<
   SideNavItemProps & { ref?: React.Ref<any> },
   'button'
 > = React.forwardRef((props: SideNavItemProps, forwardRef) => {
   const {
-    active = false,
+    active: activeProp,
     disabled = false,
     ariaCurrentValue = AriaCurrentValue.Page,
     className,
     children,
+    onClick: onClickProp,
+    glyph,
+    ...rest
   } = props;
+  const { usingKeyboard: showFocus } = useUsingKeyboardContext();
+  const active = typeof activeProp === 'boolean' && activeProp;
 
-  const rest = omit(props, [
-    'active',
-    'disabled',
-    'ariaCurrentValue',
-    'className',
-    'children',
-  ]);
+  const onClick = disabled
+    ? (e: React.MouseEvent) => {
+        e.nativeEvent.stopImmediatePropagation();
+        e.preventDefault();
+      }
+    : onClickProp;
 
-  const { usingKeyboard: showFocus } = useUsingKeyboardContext();
+  const accessibleGlyph =
+    (glyph && isComponentType(glyph, 'Glyph')) || isComponentType(glyph, 'Icon')
+      ? React.cloneElement(glyph, { 'aria-hidden': true })
```

Comment:
Does this have a performance implication, as opposed to invoking the <Glyph /> and passing the prop? (I'm not sure)

Reply:
ditto here

Reply:
Invoking the what? The component takes a rendered ReactNode.

We did research a while back to see if cloneElement had significant performance impacts, and found that wasn't the case IIRC. Even so, the number of items in navigation are not unbounded, so this doesn't have to perform well with 500 on the page (though I think it would be fine if it did).

Reply:
Omg so weird I guess GH ignores `< />` characters
`<Glyph />`